kyaml/yaml/rnode.go

```diff
@@ -1071,7 +1075,7 @@ func (rn *RNode) GetValidatedMetadata() (ResourceMeta, error) {
 		// A list doesn't require a name.
 		return m, nil
 	}
-	if m.NameMeta.Name == "" {
+	if m.NameMeta.Name == "" && m.NameMeta.GenerateName == "" {
 		return m, fmt.Errorf("missing metadata.name in object %v", m)
```

Comment:
nit: we need to change the error message if we do support this

Reply:
I was more afraid of changing the error message, giving too much hint about generateName :) I'll have a look on it though.